tests/test_sel_temperature_reader.py

```diff
@@ -0,0 +1,81 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import unittest
+
+from lsst.ts.ess.sensors.sel_temperature_reader import SelTemperature, DELIMITER
+from lsst.ts.ess.sensors.mock.mock_temperature_sensor import (
+    MockTemperatureSensor,
+    MIN_TEMP,
+    MAX_TEMP,
+)
+
+
+class SelTemperatureReaderTestCase(unittest.IsolatedAsyncioTestCase):
+    async def test_sel_temperature_reader(self):
+        num_channels = 4
+        name = "MockSensor"
+        device = MockTemperatureSensor(name, num_channels)
+        sel_temperature = SelTemperature(name, device, num_channels)
+        await sel_temperature.start()
+        await sel_temperature.read()
+        data = sel_temperature.output
+        self.assertEqual(num_channels + 3, len(data))
+        self.assertEqual(name, data[0])
+        for i in range(0, 4):
+            data_item = data[i + 3]
+            self.assertTrue(MIN_TEMP <= float(data_item) <= MAX_TEMP)
+        await sel_temperature.stop()
+
+    async def test_old_sel_temperature_reader(self):
+        num_channels = 4
+        count_offset = 1
+        name = "MockSensor"
+        device = MockTemperatureSensor(name, num_channels, count_offset)
+        sel_temperature = SelTemperature(name, device, num_channels)
+        await sel_temperature.start()
+        await sel_temperature.read()
+        data = sel_temperature.output
+        self.assertEqual(num_channels + 3, len(data))
+        self.assertEqual(name, data[0])
+        for i in range(0, 4):
+            data_item = data[i + 3]
+            self.assertTrue(MIN_TEMP <= float(data_item) <= MAX_TEMP)
+        await sel_temperature.stop()
+
+    async def test_nan_sel_temperature_reader(self):
+        num_channels = 4
+        count_offset = 1
+        nan_channel = 2
+        name = "MockSensor"
+        device = MockTemperatureSensor(
+            name, num_channels, count_offset, nan_channel=nan_channel
+        )
+        sel_temperature = SelTemperature(name, device, num_channels)
+        await sel_temperature.start()
+        await sel_temperature.read()
+        data = sel_temperature.output
+        self.assertEqual(num_channels + 3, len(data))
+        self.assertEqual(name, data[0])
+        for i in range(0, 4):
+            data_item = data[i + 3]
+            if i == nan_channel:
+                self.assertAlmostEqual(9999.999, float(data_item), 3)
```

Comment:
It is surprising that a "nan" channel returns 9999.999 instead of NaN. Surely you could make the interface to the low-level controller return NaN?

I can see where it depends what level you are testing -- if the real device returns 9999.999 for a disconnected sensor and you end up using a TCP/IP<->RS-232 adapter then you'll have to do the translation at the reading end (or in the CSC).

If you do have to use this value then please make it a public named constant (one used by, your mock controller, this test code and everywhere else that the value is read or written).